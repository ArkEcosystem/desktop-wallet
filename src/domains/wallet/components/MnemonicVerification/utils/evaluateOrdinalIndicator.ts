const ordinalPluralizer = new Intl.PluralRules("en", { type: "ordinal" });

const SUFFIXES = {
	zero: "",
	one: "st",
	two: "nd",
	few: "rd",
	other: "th",
	many: "th",
};

export const getOrdinalIndicator = (number: number) => SUFFIXES[ordinalPluralizer.select(number)];
