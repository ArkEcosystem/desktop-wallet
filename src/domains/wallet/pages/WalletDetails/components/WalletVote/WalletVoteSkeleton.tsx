import { Circle } from "app/components/Circle";
import { Skeleton } from "app/components/Skeleton";
import React from "react";

export const WalletVoteSkeleton = () => (
	<div data-testid="WalletVote__skeleton" className="flex items-center w-full">
		<Circle size="lg" shadowColor="--theme-secondary-background-color" className="mr-4 border-transparent">
			<Skeleton circle height={44} width={44} />
		</Circle>

		<div className="flex flex-col justify-between h-10">
			<Skeleton height={14} width={100} />
			<Skeleton height={16} width={150} />
		</div>

		<div className="ml-auto">
			<Skeleton height={44} width={100} />
		</div>
	</div>
);
