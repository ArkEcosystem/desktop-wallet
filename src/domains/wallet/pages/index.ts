export * from "./CreateWallet";
export * from "./ImportWallet";
export * from "./WalletDetails";
