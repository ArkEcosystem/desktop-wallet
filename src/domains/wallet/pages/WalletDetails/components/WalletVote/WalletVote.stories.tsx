import { ReadOnlyWallet } from "@arkecosystem/platform-sdk-profiles";
import { action } from "@storybook/addon-actions";
import React from "react";
import { WalletsDecorator } from "utils/storybook";

import { WalletVote } from "./WalletVote";

export default {
	title: "Domains / Wallet / Components / WalletVote",
	decorators: [
		(storyFn: any) => (
			<WalletsDecorator count={1} withDelegates={true}>
				{storyFn}
			</WalletsDecorator>
		),
	],
};

export const Default = ({ delegates }: { delegates: ReadOnlyWallet[] }) => (
	<WalletVote votes={delegates} onButtonClick={action("onButtonClick")} />
);

export const Empty = () => <WalletVote onButtonClick={action("onButtonClick")} />;

export const Loading = () => <WalletVote isLoading={true} onButtonClick={action("onButtonClick")} />;
