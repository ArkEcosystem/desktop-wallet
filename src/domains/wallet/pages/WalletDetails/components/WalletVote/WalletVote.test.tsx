import { ReadOnlyWallet, ReadWriteWallet } from "@arkecosystem/platform-sdk-profiles";
import { translations as commonTranslations } from "app/i18n/common/i18n";
import { translations as walletTranslations } from "domains/wallet/i18n";
import React from "react";
import { act, env, fireEvent, getDefaultProfileId, render, syncDelegates } from "utils/testing-library";

import { WalletVote } from "./WalletVote";

let wallet: ReadWriteWallet;

describe("WalletVote", () => {
	beforeEach(async () => {
		const profile = env.profiles().findById(getDefaultProfileId());
		wallet = profile.wallets().findById("ac38fe6d-4b67-4ef1-85be-17c5f6841129");

		await syncDelegates();
		await wallet.syncVotes();
	});

	it("should render", () => {
		const { asFragment, getByTestId } = render(<WalletVote wallet={wallet} onButtonClick={jest.fn()} />);

		expect(getByTestId("WalletVote")).toBeTruthy();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render loading state", () => {
		const { asFragment, getByTestId } = render(<WalletVote wallet={wallet} onButtonClick={jest.fn()} isLoading />);

		expect(getByTestId("WalletVote__skeleton")).toBeTruthy();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render without votes", () => {
		const walletSpy = jest.spyOn(wallet, "votes").mockReturnValue([]);

		const { asFragment, getByText } = render(<WalletVote wallet={wallet} onButtonClick={jest.fn()} />);

		expect(getByText(commonTranslations.LEARN_MORE)).toBeTruthy();
		expect(asFragment()).toMatchSnapshot();

		walletSpy.mockRestore();
	});

	it("should render the maximum votes", () => {
		const walletSpy = jest.spyOn(wallet, "votes").mockReturnValue([]);
		const maxVotesSpy = jest.spyOn(wallet.network(), "maximumVotesPerWallet").mockReturnValue(101);

		const { asFragment, getByText } = render(<WalletVote wallet={wallet} onButtonClick={jest.fn()} />);

		expect(getByText("0/101")).toBeTruthy();
		expect(asFragment()).toMatchSnapshot();

		walletSpy.mockRestore();
		maxVotesSpy.mockRestore();
	});

	describe("single vote networks", () => {
		it("should render a vote for an active delegate", () => {
			const walletSpy = jest.spyOn(wallet, "votes").mockReturnValue([
				new ReadOnlyWallet({
					address: wallet.address(),
					explorerLink: "",
					publicKey: wallet.publicKey(),
					username: "arkx",
					rank: 10,
				}),
			]);

			const { asFragment, getByText } = render(<WalletVote wallet={wallet} onButtonClick={jest.fn()} />);

			const delegate = wallet.votes()[0];

			expect(getByText(delegate.username())).toBeTruthy();
			expect(getByText(`#${delegate.rank()}`)).toBeTruthy();
			expect(getByText(walletTranslations.PAGE_WALLET_DETAILS.VOTES.ACTIVE)).toBeTruthy();

			expect(asFragment()).toMatchSnapshot();

			walletSpy.mockRestore();
		});

		it("should render a vote for a standby delegate", () => {
			const walletSpy = jest.spyOn(wallet, "votes").mockReturnValue([
				new ReadOnlyWallet({
					address: wallet.address(),
					explorerLink: "",
					publicKey: wallet.publicKey(),
					username: "arkx",
					rank: 52,
				}),
			]);

			const { asFragment, getByText } = render(<WalletVote wallet={wallet} onButtonClick={jest.fn()} />);

			const delegate = wallet.votes()[0];

			expect(getByText(delegate.username())).toBeTruthy();
			expect(getByText(`#${delegate.rank()}`)).toBeTruthy();
			expect(getByText(walletTranslations.PAGE_WALLET_DETAILS.VOTES.STANDBY)).toBeTruthy();

			expect(asFragment()).toMatchSnapshot();

			walletSpy.mockRestore();
		});

		it("should render a vote for a delegate without rank", () => {
			const walletSpy = jest.spyOn(wallet, "votes").mockReturnValue([
				new ReadOnlyWallet({
					address: wallet.address(),
					explorerLink: "",
					publicKey: wallet.publicKey(),
					username: "arkx",
				}),
			]);

			const { asFragment, getByText } = render(<WalletVote wallet={wallet} onButtonClick={jest.fn()} />);

			const delegate = wallet.votes()[0];

			expect(getByText(delegate.username())).toBeTruthy();
			expect(getByText(commonTranslations.NOT_AVAILABLE)).toBeTruthy();
			expect(getByText(walletTranslations.PAGE_WALLET_DETAILS.VOTES.STANDBY)).toBeTruthy();

			expect(getByText("information-circle.svg")).toBeTruthy();
			expect(asFragment()).toMatchSnapshot();

			walletSpy.mockRestore();
		});
	});

	describe("multi vote networks", () => {
		let maxVotesSpy: jest.SpyInstance;

		beforeEach(() => (maxVotesSpy = jest.spyOn(wallet.network(), "maximumVotesPerWallet").mockReturnValue(101)));

		afterEach(() => maxVotesSpy.mockRestore());

		it("should render a vote for multiple active delegates", () => {
			const walletSpy = jest.spyOn(wallet, "votes").mockReturnValue([
				new ReadOnlyWallet({
					address: wallet.address(),
					explorerLink: "",
					publicKey: wallet.publicKey(),
					username: "arkx",
					rank: 1,
				}),
				new ReadOnlyWallet({
					address: wallet.address(),
					explorerLink: "",
					publicKey: wallet.publicKey(),
					username: "arky",
					rank: 2,
				}),
			]);

			const { asFragment, getByText } = render(<WalletVote wallet={wallet} onButtonClick={jest.fn()} />);

			expect(getByText(walletTranslations.PAGE_WALLET_DETAILS.VOTES.MULTIVOTE)).toBeTruthy();
			expect(getByText(walletTranslations.PAGE_WALLET_DETAILS.VOTES.ACTIVE_plural)).toBeTruthy();

			expect(asFragment()).toMatchSnapshot();

			walletSpy.mockRestore();
		});

		it("should render a vote for multiple active delegates", () => {
			const walletSpy = jest.spyOn(wallet, "votes").mockReturnValue([
				new ReadOnlyWallet({
					address: wallet.address(),
					explorerLink: "",
					publicKey: wallet.publicKey(),
					username: "arkx",
				}),
				new ReadOnlyWallet({
					address: wallet.address(),
					explorerLink: "",
					publicKey: wallet.publicKey(),
					username: "arky",
				}),
			]);

			const { asFragment, getByText } = render(<WalletVote wallet={wallet} onButtonClick={jest.fn()} />);

			expect(getByText(walletTranslations.PAGE_WALLET_DETAILS.VOTES.MULTIVOTE)).toBeTruthy();
			expect(getByText(walletTranslations.PAGE_WALLET_DETAILS.VOTES.STANDBY_plural)).toBeTruthy();

			expect(asFragment()).toMatchSnapshot();

			walletSpy.mockRestore();
		});

		it("should render a vote for multiple active and standby delegates", () => {
			const walletSpy = jest.spyOn(wallet, "votes").mockReturnValue([
				new ReadOnlyWallet({
					address: wallet.address(),
					explorerLink: "",
					publicKey: wallet.publicKey(),
					username: "arkx",
					rank: 1,
				}),
				new ReadOnlyWallet({
					address: wallet.address(),
					explorerLink: "",
					publicKey: wallet.publicKey(),
					username: "arky",
				}),
			]);

			const { asFragment, getByText } = render(<WalletVote wallet={wallet} onButtonClick={jest.fn()} />);

			expect(getByText(walletTranslations.PAGE_WALLET_DETAILS.VOTES.MULTIVOTE)).toBeTruthy();
			expect(getByText("Active 1")).toBeTruthy();
			expect(getByText("/ Standby 1")).toBeTruthy();

			expect(getByText("information-circle.svg")).toBeTruthy();
			expect(asFragment()).toMatchSnapshot();

			walletSpy.mockRestore();
		});
	});

	it("should emit action on multivote click", () => {
		const walletSpy = jest.spyOn(wallet, "votes").mockReturnValue([
			new ReadOnlyWallet({
				address: wallet.address(),
				explorerLink: "",
				publicKey: wallet.publicKey(),
				username: "arkx",
				rank: 1,
			}),
			new ReadOnlyWallet({
				address: wallet.address(),
				explorerLink: "",
				publicKey: wallet.publicKey(),
				username: "arky",
				rank: 2,
			}),
		]);

		const onButtonClick = jest.fn();

		const { getByText } = render(<WalletVote wallet={wallet} onButtonClick={onButtonClick} />);

		act(() => {
			fireEvent.click(getByText(walletTranslations.PAGE_WALLET_DETAILS.VOTES.MULTIVOTE));
		});

		expect(onButtonClick).toHaveBeenCalled();

		walletSpy.mockRestore();
	});

	it("should emit action on button click", () => {
		const walletSpy = jest.spyOn(wallet, "votes").mockReturnValue([]);

		const onButtonClick = jest.fn();

		const { getByText } = render(<WalletVote wallet={wallet} onButtonClick={onButtonClick} />);

		act(() => {
			fireEvent.click(getByText(commonTranslations.VOTE));
		});

		expect(onButtonClick).toHaveBeenCalled();

		walletSpy.mockRestore();
	});
});
