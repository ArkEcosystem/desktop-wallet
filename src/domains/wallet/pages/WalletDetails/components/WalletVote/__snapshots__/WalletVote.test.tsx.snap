// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletVote multi vote networks should render a vote for multiple active and standby delegates 1`] = `
<DocumentFragment>
  <div
    class="flex items-center w-full"
    data-testid="WalletVote"
  >
    <div
      class="Circle-uw7h4j-0 kUEiXZ border-theme-secondary-900 dark:border-theme-secondary-700 text-theme-secondary-900 dark:text-theme-secondary-700"
    >
      <div
        class="sc-bdfBwQ fdjKgV"
        height="17"
        width="17"
      >
        <svg>
          vote.svg
        </svg>
      </div>
    </div>
    <div
      class="flex flex-1 ml-4"
    >
      <div
        class="flex flex-col justify-between font-semibold"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700"
        >
          Voting for
        </span>
        <span
          class="transition-colors duration-200 cursor-pointer text-theme-primary-600 hover:text-theme-primary-700 active:text-theme-primary-500"
        >
          Multivote
        </span>
      </div>
      <div
        class="flex flex-col justify-between pl-6 ml-6 font-semibold border-l border-theme-secondary-300 dark:border-theme-secondary-800"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700"
        >
          Votes
        </span>
        <span>
          2
          <span
            class="text-theme-secondary-500 dark:text-theme-secondary-700"
          >
            /101
          </span>
        </span>
      </div>
    </div>
    <div
      class="flex flex-col justify-between items-end pr-6 mr-6 font-semibold border-r border-theme-secondary-300 dark:border-theme-secondary-800"
    >
      <span
        class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700 "
      >
        Delegate Status
      </span>
      <div
        class="flex justify-end items-center space-x-2"
      >
        <span>
          <div
            class="sc-bdfBwQ iyevwF text-theme-primary-300 dark:text-theme-secondary-600"
            height="20"
            width="20"
          >
            <svg>
              information-circle.svg
            </svg>
          </div>
        </span>
        <span
          class="font-semibold"
        >
          Active 1
          <span
            class="text-theme-secondary-500 dark:text-theme-secondary-700"
          >
             / Standby 1
          </span>
        </span>
      </div>
    </div>
    <button
      class="Button__StyledButton-mdtvtf-0 iUSBMV space-x-2"
      data-testid="WalletVote__button"
      type="button"
    >
      <div
        class="flex relative items-center space-x-2"
      >
        <div
          class="sc-bdfBwQ fdjKgV"
          height="17"
          width="17"
        >
          <svg>
            vote.svg
          </svg>
        </div>
        <span>
          Vote
        </span>
      </div>
    </button>
  </div>
</DocumentFragment>
`;

exports[`WalletVote multi vote networks should render a vote for multiple active delegates 1`] = `
<DocumentFragment>
  <div
    class="flex items-center w-full"
    data-testid="WalletVote"
  >
    <div
      class="Circle-uw7h4j-0 kUEiXZ border-theme-secondary-900 dark:border-theme-secondary-700 text-theme-secondary-900 dark:text-theme-secondary-700"
    >
      <div
        class="sc-bdfBwQ fdjKgV"
        height="17"
        width="17"
      >
        <svg>
          vote.svg
        </svg>
      </div>
    </div>
    <div
      class="flex flex-1 ml-4"
    >
      <div
        class="flex flex-col justify-between font-semibold"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700"
        >
          Voting for
        </span>
        <span
          class="transition-colors duration-200 cursor-pointer text-theme-primary-600 hover:text-theme-primary-700 active:text-theme-primary-500"
        >
          Multivote
        </span>
      </div>
      <div
        class="flex flex-col justify-between pl-6 ml-6 font-semibold border-l border-theme-secondary-300 dark:border-theme-secondary-800"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700"
        >
          Votes
        </span>
        <span>
          2
          <span
            class="text-theme-secondary-500 dark:text-theme-secondary-700"
          >
            /101
          </span>
        </span>
      </div>
    </div>
    <div
      class="flex flex-col justify-between items-end pr-6 mr-6 font-semibold border-r border-theme-secondary-300 dark:border-theme-secondary-800"
    >
      <span
        class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700 "
      >
        Delegate Status
      </span>
      <div
        class="flex justify-end items-center space-x-2"
      >
        <span
          class="font-semibold text-theme-success-600"
        >
          All Active
        </span>
      </div>
    </div>
    <button
      class="Button__StyledButton-mdtvtf-0 iUSBMV space-x-2"
      data-testid="WalletVote__button"
      type="button"
    >
      <div
        class="flex relative items-center space-x-2"
      >
        <div
          class="sc-bdfBwQ fdjKgV"
          height="17"
          width="17"
        >
          <svg>
            vote.svg
          </svg>
        </div>
        <span>
          Vote
        </span>
      </div>
    </button>
  </div>
</DocumentFragment>
`;

exports[`WalletVote multi vote networks should render a vote for multiple active delegates 2`] = `
<DocumentFragment>
  <div
    class="flex items-center w-full"
    data-testid="WalletVote"
  >
    <div
      class="Circle-uw7h4j-0 kUEiXZ border-theme-secondary-900 dark:border-theme-secondary-700 text-theme-secondary-900 dark:text-theme-secondary-700"
    >
      <div
        class="sc-bdfBwQ fdjKgV"
        height="17"
        width="17"
      >
        <svg>
          vote.svg
        </svg>
      </div>
    </div>
    <div
      class="flex flex-1 ml-4"
    >
      <div
        class="flex flex-col justify-between font-semibold"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700"
        >
          Voting for
        </span>
        <span
          class="transition-colors duration-200 cursor-pointer text-theme-primary-600 hover:text-theme-primary-700 active:text-theme-primary-500"
        >
          Multivote
        </span>
      </div>
      <div
        class="flex flex-col justify-between pl-6 ml-6 font-semibold border-l border-theme-secondary-300 dark:border-theme-secondary-800"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700"
        >
          Votes
        </span>
        <span>
          2
          <span
            class="text-theme-secondary-500 dark:text-theme-secondary-700"
          >
            /101
          </span>
        </span>
      </div>
    </div>
    <div
      class="flex flex-col justify-between items-end pr-6 mr-6 font-semibold border-r border-theme-secondary-300 dark:border-theme-secondary-800"
    >
      <span
        class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700 "
      >
        Delegate Status
      </span>
      <div
        class="flex justify-end items-center space-x-2"
      >
        <span>
          <div
            class="sc-bdfBwQ iyevwF text-theme-primary-300 dark:text-theme-secondary-600"
            height="20"
            width="20"
          >
            <svg>
              information-circle.svg
            </svg>
          </div>
        </span>
        <span
          class="font-semibold text-theme-warning-500"
        >
          All Standby
        </span>
      </div>
    </div>
    <button
      class="Button__StyledButton-mdtvtf-0 iUSBMV space-x-2"
      data-testid="WalletVote__button"
      type="button"
    >
      <div
        class="flex relative items-center space-x-2"
      >
        <div
          class="sc-bdfBwQ fdjKgV"
          height="17"
          width="17"
        >
          <svg>
            vote.svg
          </svg>
        </div>
        <span>
          Vote
        </span>
      </div>
    </button>
  </div>
</DocumentFragment>
`;

exports[`WalletVote should render 1`] = `
<DocumentFragment>
  <div
    class="flex items-center w-full"
    data-testid="WalletVote"
  >
    <div
      class="Circle-uw7h4j-0 kUEiXZ border-theme-secondary-900 dark:border-theme-secondary-700 text-theme-secondary-900 dark:text-theme-secondary-700"
    >
      <div
        class="sc-bdfBwQ fdjKgV"
        height="17"
        width="17"
      >
        <svg>
          vote.svg
        </svg>
      </div>
    </div>
    <div
      class="flex flex-1 ml-4"
    >
      <div
        class="flex flex-col justify-between font-semibold"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700"
        >
          Voting for
        </span>
        <span>
          arkx
        </span>
      </div>
      <div
        class="flex flex-col justify-between pl-6 ml-6 font-semibold border-l border-theme-secondary-300 dark:border-theme-secondary-800"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700"
        >
          Rank
        </span>
        <span>
          N/A
        </span>
      </div>
    </div>
    <div
      class="flex flex-col justify-between items-end pr-6 mr-6 font-semibold border-r border-theme-secondary-300 dark:border-theme-secondary-800"
    >
      <span
        class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700 "
      >
        Delegate Status
      </span>
      <div
        class="flex justify-end items-center space-x-2"
      >
        <span>
          <div
            class="sc-bdfBwQ iyevwF text-theme-primary-300 dark:text-theme-secondary-600"
            height="20"
            width="20"
          >
            <svg>
              information-circle.svg
            </svg>
          </div>
        </span>
        <span
          class="font-semibold text-theme-warning-500"
        >
          Standby
        </span>
      </div>
    </div>
    <button
      class="Button__StyledButton-mdtvtf-0 iUSBMV space-x-2"
      data-testid="WalletVote__button"
      type="button"
    >
      <div
        class="flex relative items-center space-x-2"
      >
        <div
          class="sc-bdfBwQ fdjKgV"
          height="17"
          width="17"
        >
          <svg>
            vote.svg
          </svg>
        </div>
        <span>
          Vote
        </span>
      </div>
    </button>
  </div>
</DocumentFragment>
`;

exports[`WalletVote should render loading state 1`] = `
<DocumentFragment>
  <div
    class="flex items-center w-full"
    data-testid="WalletVote__skeleton"
  >
    <div
      class="Circle-uw7h4j-0 kUEiXZ mr-4 border-transparent"
    >
      <span>
        <span
          class="flex items-center leading-none"
        >
          <span
            class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
            style="width: 44px; height: 44px; border-radius: 50%;"
          >
            ‌
          </span>
          ‌
        </span>
      </span>
    </div>
    <div
      class="flex flex-col justify-between h-10"
    >
      <span>
        <span
          class="flex items-center leading-none"
        >
          <span
            class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
            style="width: 100px; height: 14px;"
          >
            ‌
          </span>
          ‌
        </span>
      </span>
      <span>
        <span
          class="flex items-center leading-none"
        >
          <span
            class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
            style="width: 150px; height: 16px;"
          >
            ‌
          </span>
          ‌
        </span>
      </span>
    </div>
    <div
      class="ml-auto"
    >
      <span>
        <span
          class="flex items-center leading-none"
        >
          <span
            class="react-loading-skeleton css-1vmnjpn-skeletonStyles-Skeleton"
            style="width: 100px; height: 44px;"
          >
            ‌
          </span>
          ‌
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`WalletVote should render the maximum votes 1`] = `
<DocumentFragment>
  <div
    class="flex items-center w-full"
    data-testid="WalletVote"
  >
    <div
      class="Circle-uw7h4j-0 kUEiXZ border-theme-secondary-500 dark:border-theme-secondary-700 text-theme-secondary-500 dark:text-theme-secondary-700"
    >
      <div
        class="sc-bdfBwQ fdjKgV"
        height="17"
        width="17"
      >
        <svg>
          vote.svg
        </svg>
      </div>
    </div>
    <div
      class="flex flex-1 ml-4"
    >
      <div
        class="flex flex-col justify-between"
      >
        <span
          class="font-semibold"
        >
          My Votes
          <span
            class="ml-1 text-theme-secondary-500 dark:text-theme-secondary-700"
          >
            0/101
          </span>
        </span>
        <span
          class="leading-none"
        >
          <span
            class="mr-1 text-sm text-theme-secondary-500 dark:text-theme-secondary-700"
          >
            You have not yet voted for a delegate
          </span>
          <a
            class="Link__AnchorStyled-sc-1q7sbxr-0 cIuuRE inline-flex items-center font-semibold transition-colors duration-200 cursor-pointer text-theme-primary-600 hover:text-theme-primary-700 hover:underline active:text-theme-primary-500"
            data-testid="Link"
            rel="noopener noreferrer"
            to="https://ark.dev/docs/desktop-wallet/user-guides/how-to-vote-unvote"
          >
            <span
              class="underline-dotted"
            >
              <span
                class="text-sm"
              >
                Learn more
              </span>
            </span>
            <div
              class="sc-bdfBwQ fPrsah flex-shrink-0 ml-2 text-sm"
              data-testid="Link__external"
              height="1em"
              width="1em"
            >
              <svg>
                redirect.svg
              </svg>
            </div>
          </a>
        </span>
      </div>
    </div>
    <button
      class="Button__StyledButton-mdtvtf-0 iUSBMV space-x-2"
      data-testid="WalletVote__button"
      type="button"
    >
      <div
        class="flex relative items-center space-x-2"
      >
        <div
          class="sc-bdfBwQ fdjKgV"
          height="17"
          width="17"
        >
          <svg>
            vote.svg
          </svg>
        </div>
        <span>
          Vote
        </span>
      </div>
    </button>
  </div>
</DocumentFragment>
`;

exports[`WalletVote should render without votes 1`] = `
<DocumentFragment>
  <div
    class="flex items-center w-full"
    data-testid="WalletVote"
  >
    <div
      class="Circle-uw7h4j-0 kUEiXZ border-theme-secondary-500 dark:border-theme-secondary-700 text-theme-secondary-500 dark:text-theme-secondary-700"
    >
      <div
        class="sc-bdfBwQ fdjKgV"
        height="17"
        width="17"
      >
        <svg>
          vote.svg
        </svg>
      </div>
    </div>
    <div
      class="flex flex-1 ml-4"
    >
      <div
        class="flex flex-col justify-between"
      >
        <span
          class="font-semibold"
        >
          My Vote
          <span
            class="ml-1 text-theme-secondary-500 dark:text-theme-secondary-700"
          >
            0/1
          </span>
        </span>
        <span
          class="leading-none"
        >
          <span
            class="mr-1 text-sm text-theme-secondary-500 dark:text-theme-secondary-700"
          >
            You have not yet voted for a delegate
          </span>
          <a
            class="Link__AnchorStyled-sc-1q7sbxr-0 cIuuRE inline-flex items-center font-semibold transition-colors duration-200 cursor-pointer text-theme-primary-600 hover:text-theme-primary-700 hover:underline active:text-theme-primary-500"
            data-testid="Link"
            rel="noopener noreferrer"
            to="https://ark.dev/docs/desktop-wallet/user-guides/how-to-vote-unvote"
          >
            <span
              class="underline-dotted"
            >
              <span
                class="text-sm"
              >
                Learn more
              </span>
            </span>
            <div
              class="sc-bdfBwQ fPrsah flex-shrink-0 ml-2 text-sm"
              data-testid="Link__external"
              height="1em"
              width="1em"
            >
              <svg>
                redirect.svg
              </svg>
            </div>
          </a>
        </span>
      </div>
    </div>
    <button
      class="Button__StyledButton-mdtvtf-0 iUSBMV space-x-2"
      data-testid="WalletVote__button"
      type="button"
    >
      <div
        class="flex relative items-center space-x-2"
      >
        <div
          class="sc-bdfBwQ fdjKgV"
          height="17"
          width="17"
        >
          <svg>
            vote.svg
          </svg>
        </div>
        <span>
          Vote
        </span>
      </div>
    </button>
  </div>
</DocumentFragment>
`;

exports[`WalletVote single vote networks should render a vote for a delegate without rank 1`] = `
<DocumentFragment>
  <div
    class="flex items-center w-full"
    data-testid="WalletVote"
  >
    <div
      class="Circle-uw7h4j-0 kUEiXZ border-theme-secondary-900 dark:border-theme-secondary-700 text-theme-secondary-900 dark:text-theme-secondary-700"
    >
      <div
        class="sc-bdfBwQ fdjKgV"
        height="17"
        width="17"
      >
        <svg>
          vote.svg
        </svg>
      </div>
    </div>
    <div
      class="flex flex-1 ml-4"
    >
      <div
        class="flex flex-col justify-between font-semibold"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700"
        >
          Voting for
        </span>
        <span>
          arkx
        </span>
      </div>
      <div
        class="flex flex-col justify-between pl-6 ml-6 font-semibold border-l border-theme-secondary-300 dark:border-theme-secondary-800"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700"
        >
          Rank
        </span>
        <span>
          N/A
        </span>
      </div>
    </div>
    <div
      class="flex flex-col justify-between items-end pr-6 mr-6 font-semibold border-r border-theme-secondary-300 dark:border-theme-secondary-800"
    >
      <span
        class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700 "
      >
        Delegate Status
      </span>
      <div
        class="flex justify-end items-center space-x-2"
      >
        <span>
          <div
            class="sc-bdfBwQ iyevwF text-theme-primary-300 dark:text-theme-secondary-600"
            height="20"
            width="20"
          >
            <svg>
              information-circle.svg
            </svg>
          </div>
        </span>
        <span
          class="font-semibold text-theme-warning-500"
        >
          Standby
        </span>
      </div>
    </div>
    <button
      class="Button__StyledButton-mdtvtf-0 iUSBMV space-x-2"
      data-testid="WalletVote__button"
      type="button"
    >
      <div
        class="flex relative items-center space-x-2"
      >
        <div
          class="sc-bdfBwQ fdjKgV"
          height="17"
          width="17"
        >
          <svg>
            vote.svg
          </svg>
        </div>
        <span>
          Vote
        </span>
      </div>
    </button>
  </div>
</DocumentFragment>
`;

exports[`WalletVote single vote networks should render a vote for a standby delegate 1`] = `
<DocumentFragment>
  <div
    class="flex items-center w-full"
    data-testid="WalletVote"
  >
    <div
      class="Circle-uw7h4j-0 kUEiXZ border-theme-secondary-900 dark:border-theme-secondary-700 text-theme-secondary-900 dark:text-theme-secondary-700"
    >
      <div
        class="sc-bdfBwQ fdjKgV"
        height="17"
        width="17"
      >
        <svg>
          vote.svg
        </svg>
      </div>
    </div>
    <div
      class="flex flex-1 ml-4"
    >
      <div
        class="flex flex-col justify-between font-semibold"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700"
        >
          Voting for
        </span>
        <span>
          arkx
        </span>
      </div>
      <div
        class="flex flex-col justify-between pl-6 ml-6 font-semibold border-l border-theme-secondary-300 dark:border-theme-secondary-800"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700"
        >
          Rank
        </span>
        <span>
          #52
        </span>
      </div>
    </div>
    <div
      class="flex flex-col justify-between items-end pr-6 mr-6 font-semibold border-r border-theme-secondary-300 dark:border-theme-secondary-800"
    >
      <span
        class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700 "
      >
        Delegate Status
      </span>
      <div
        class="flex justify-end items-center space-x-2"
      >
        <span>
          <div
            class="sc-bdfBwQ iyevwF text-theme-primary-300 dark:text-theme-secondary-600"
            height="20"
            width="20"
          >
            <svg>
              information-circle.svg
            </svg>
          </div>
        </span>
        <span
          class="font-semibold text-theme-warning-500"
        >
          Standby
        </span>
      </div>
    </div>
    <button
      class="Button__StyledButton-mdtvtf-0 iUSBMV space-x-2"
      data-testid="WalletVote__button"
      type="button"
    >
      <div
        class="flex relative items-center space-x-2"
      >
        <div
          class="sc-bdfBwQ fdjKgV"
          height="17"
          width="17"
        >
          <svg>
            vote.svg
          </svg>
        </div>
        <span>
          Vote
        </span>
      </div>
    </button>
  </div>
</DocumentFragment>
`;

exports[`WalletVote single vote networks should render a vote for an active delegate 1`] = `
<DocumentFragment>
  <div
    class="flex items-center w-full"
    data-testid="WalletVote"
  >
    <div
      class="Circle-uw7h4j-0 kUEiXZ border-theme-secondary-900 dark:border-theme-secondary-700 text-theme-secondary-900 dark:text-theme-secondary-700"
    >
      <div
        class="sc-bdfBwQ fdjKgV"
        height="17"
        width="17"
      >
        <svg>
          vote.svg
        </svg>
      </div>
    </div>
    <div
      class="flex flex-1 ml-4"
    >
      <div
        class="flex flex-col justify-between font-semibold"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700"
        >
          Voting for
        </span>
        <span>
          arkx
        </span>
      </div>
      <div
        class="flex flex-col justify-between pl-6 ml-6 font-semibold border-l border-theme-secondary-300 dark:border-theme-secondary-800"
      >
        <span
          class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700"
        >
          Rank
        </span>
        <span>
          #10
        </span>
      </div>
    </div>
    <div
      class="flex flex-col justify-between items-end pr-6 mr-6 font-semibold border-r border-theme-secondary-300 dark:border-theme-secondary-800"
    >
      <span
        class="text-sm text-theme-secondary-500 dark:text-theme-secondary-700 "
      >
        Delegate Status
      </span>
      <div
        class="flex justify-end items-center space-x-2"
      >
        <span
          class="font-semibold text-theme-success-600"
        >
          Active
        </span>
      </div>
    </div>
    <button
      class="Button__StyledButton-mdtvtf-0 iUSBMV space-x-2"
      data-testid="WalletVote__button"
      type="button"
    >
      <div
        class="flex relative items-center space-x-2"
      >
        <div
          class="sc-bdfBwQ fdjKgV"
          height="17"
          width="17"
        >
          <svg>
            vote.svg
          </svg>
        </div>
        <span>
          Vote
        </span>
      </div>
    </button>
  </div>
</DocumentFragment>
`;
